### DISCUSSION TITLE
Could you provide a Julia wrapper for SU2?

### DISCUSSION THREAD

**Discussion by MrBenzWorld** [su2code/SU2#1899] [Ideas]
There is no proper CFD package available in Julia language.  you have a python wrapper. could you please add a wrapper in Julia for SU2?

**Comment by Nicola-Fonzi** [comment]
Hi @MrBenzWorld,

Thank you for posting this. I was looking at Julia a couple of weeks ago, but had no time to investigate it further yet. Do you know of other similar software that has been wrapped? If yes, can you point me to that?

**Reply by MrBenzWorld** [reply]
it is good for SU2 also because preprocessing and post-processing can be done in Julia in a better way.
openfoam, code-saturne, and SU2 are Open-Source CFD Codes. but SU2 has a python wrapper.
so, instead of writing all code in Julia, it is better to write a wrapper for SU2 in Julia. Julia language was also developed for numerical methods but no CFD toolbox is available yet.

finally, SU2 + Julia is the best solution. SU2 can develop post-processing tools using Julia.
I am also a beginner at learning the Julia language. trying to find the CFD package, I think only SU2 can solve this problem.

Julia is developed at MIT, and SU2 at Stanford University, but both don't find communication on it.

Please help.

@ablaom, @KristofferC @ViralBShah @logankilpatrick @JuliaLang  @Nicola-Fonzi

**Comment by bigfooted** [comment]
Hi,
As with all open source codes, the SU2 code can be extended by any developer, and the SU2 code organically develops following the needs of the community. If you, as part of the community, see a need for a Julia wrapper for SU2, I encourage you to download the SU2 code, write the Julia wrapper in a way that you seem fit, and create a pull request for your feature branch. We would be happy to bring your efforts into SU2.
Here are some links to get you started:
https://su2code.github.io/docs_v7/Build-SU2-Linux-MacOS/
https://discourse.julialang.org/t/best-way-to-create-a-julia-function-wrapper-of-a-cpp-function/25915
https://blog.esciencecenter.nl/10-examples-of-embedding-julia-in-c-c-66282477e62c
Have fun, and I'm looking forward to your contributions!

**Reply by MrBenzWorld** [reply]
Thank you very much for your response. I have just started learning Julia for CFD.  I will try it eventually.
